inspect_byproducts.py

```diff
@@ -0,0 +1,176 @@
+"""
+
+<Program Name>
+  inspect_byproducts.py
+
+<Author>
+  Sachit Malik <i.sachitmalik@gmail.com>
+
+<Started>
+  June 6, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Requires>
+  in-toto - installed on the path
+
+<Purpose>
+  Inspections constitute an important part of in-toto. The script take the path
+  of a link file, a random string and an operator, which is used to compare
+  a certain property of the link file with the random string.
+
+  Suppose the link file is located at /user/abc/def/package.45gh325.link
+  and the user wants to check whether for the corresponding step (hence the link file),
+  the corresponding stderr field contains the string "test".
+
+  The usage would be as follows:
+  python inspect_byproducts.py -l  /user/abc/def/package.45gh325.link  -st stderr -o contains test
+
+  General usage:
+  python inspect_byproducts.py -l  <path/to/link/metadata/file>  -st <stdout|stderr>
+    -o [ is | is not | contains | contains not] <string to be tested>
+
+
+"""
+
+import os
+import sys
+import argparse
+import in_toto.log
+from in_toto.models.link import Link as link_import
+import securesystemslib.exceptions
+
+
+def inspect_byproducts(link, std, operator, inputstring):
+    """
+
+    <Purpose>
+    A function which performs the inspection as described above depending on
+    various arguments.
+
+    <Arguments>
+      link:
+        the path to the link file
+
+      std:
+        whether to check stdout or stderr field
+
+      operator:
+        is | is not | contains | contains not
+
+     inputstring:
+        the string to be checked
+
+    <Exceptions>
+      Raises KeyError, in case the field corresponding to the key
+        in the dictionary is empty
+
+    <Returns>
+      Integer
+        0 - if the inspection is successful
+        1 - if the inspection is unsuccessful
+        2 - if the user supplied arguments are invalid
+
+
+
+    """
+    if not os.path.exists(link):
+        print("The path to the link file is invalid")
+        return 2
+
+    else:
+      l = link_import.read_from_file(link)
+
+      try:
+        std_out_err = l.byproducts[std]
+      except KeyError:
+        raise KeyError("The field corresponding to " + std + " in the link "
+                                                             "file is empty.")
+
+      switch_dict = {'is': (std_out_err == inputstring),
```

Comment:
Alright. So i changed it from dictionary to if conditions. No dictionary required now. Thanks! The changes will be reflected in the next commit.